DPPP/test/unit/tBDABuffer.cc

```diff
@@ -0,0 +1,391 @@
+    BOOST_CHECK(!buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, nullptr, nullptr, flags2));
+    BOOST_CHECK_EQUAL(buffer.GetNumberOfElements(), size_t {n_channels * n_correlations});
+    BOOST_CHECK_EQUAL(*buffer.GetFullResFlags(), flags[0]);
+    BOOST_CHECK_EQUAL(*(buffer.GetFullResFlags() + 1), flags2[0]);
+}
+
+BOOST_AUTO_TEST_CASE( test_bda_uvw )
+{
+    const size_t n_channels {2};
+    const size_t n_correlations {3};
+
+    BDABuffer buffer {2 * n_channels * n_correlations};
+    const double uvw[] {1, 2, 3};
+    const double uvw2[] {3, 4, 5};
+    const double uvw3[] {5, 6, 7};
+
+    BOOST_CHECK(buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, nullptr, nullptr, nullptr, uvw));
+    BOOST_CHECK(buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, nullptr, nullptr, nullptr, uvw2));
+    BOOST_CHECK(!buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, nullptr, nullptr, nullptr, uvw3));
+    BOOST_CHECK_EQUAL(buffer.GetNumberOfElements(), size_t {2 * n_channels * n_correlations});
+    const std::vector<BDABuffer::Row> rows = buffer.GetRows();
+    BOOST_CHECK_EQUAL(rows[0].GetDataSize(), n_channels * n_correlations);
+    BOOST_CHECK_EQUAL(rows[0].uvw_[0], uvw[0]);
+    BOOST_CHECK_EQUAL(rows[0].uvw_[1], uvw[1]);
+    BOOST_CHECK_EQUAL(rows[1].uvw_[0], uvw2[0]);
+    BOOST_CHECK_EQUAL(rows[1].uvw_[1], uvw2[1]);
+}
+
+BOOST_AUTO_TEST_CASE( test_bda_clear )
+{
+    const size_t n_channels {2};
+    const size_t n_correlations {3};
+    BDABuffer buffer {3 * n_channels * n_correlations + 1};
+    const bool flags[] {true};
+    const bool fr_flags[] {true, true};
+    const std::complex<float> data {1};
+    const float weights[] {1};
+
+    BOOST_CHECK(buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, &data));
+    BOOST_CHECK(buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, flags));
+    BOOST_CHECK(buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, nullptr, weights, fr_flags));
+    BOOST_CHECK(!buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, &data));
+    BOOST_CHECK_EQUAL(buffer.GetNumberOfElements(), size_t {3 * n_channels * n_correlations});
+    buffer.Clear();
+    BOOST_CHECK_EQUAL(buffer.GetNumberOfElements(), size_t {0});
+}
+
+BOOST_AUTO_TEST_CASE( test_get_rows )
+{
+    const size_t n_channels {2};
+    const size_t n_correlations {3};
+    BDABuffer buffer {3 * n_channels * n_correlations + 1};
+    const bool flags[] {true};
+    const bool fr_flags[] {false};
+    const std::complex<float> data {1};
+    const float weights[] {1};
+
+    buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, &data, flags, weights, fr_flags);
+
+    BOOST_CHECK_EQUAL(*buffer.GetData(size_t {0}), data);
+    BOOST_CHECK_EQUAL(*buffer.GetFlags(size_t {0}), flags[0]);
+    BOOST_CHECK_EQUAL(*buffer.GetFullResFlags(size_t {0}), fr_flags[0]);
+    BOOST_CHECK_EQUAL(*buffer.GetWeights(size_t {0}), weights[0]);
```

Comment:
- GetXXX() may return nullptr, so check it.
- Check all elements instead of only the first one.
- Is the size_t cast really necessary here? Using 0 or 0u should work I think.

Reply:
Similar as above, I think this is overkill since the test should fail anyway, unless it is a c++ thing that must be done.